src/checkqueue.h

```diff
@@ -155,6 +141,13 @@ class CCheckQueue
         }
     }
 
+    // Since this class manages its own resources, which is a thread
+    // pool `m_worker_threads`, copy operations are not appropriate.
+    CCheckQueue(const CCheckQueue&) = delete;
+    CCheckQueue& operator=(const CCheckQueue&) = delete;
+    CCheckQueue(CCheckQueue&&) = default;
```

Comment:
I'd explicitly set the move operations to `= delete`. I think `std::mutex` is  not movable so these will be deleted anyways, but it feels safer to just mark them as deleted too. There will be threads running throughout the lifetime of `CCheckQueue`, so moving the object while something might be going on or threads waiting on a lock wouldn't be safe

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/26762#issuecomment-1500980698).